final_project/flaskr/templates/scripts.js

```diff
@@ -499,7 +481,7 @@ BikeMap.prototype.updateRackStatus = function(rack_id) {
 
 
 BikeMap.prototype.vote = function(e) {
-    
+    console.log('voting');
     if (!this.auth.currentUser) {
         // redirect user to sign in
         bikemap.signIn()
```

Comment:
If the whole rest of this function is in the `else` part then you don't need `else` at all.

```
if (!this.auth.currentUser) {
  bikemap.signIn();
}

let rack_id = ....
```